BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.m

```diff
@@ -0,0 +1,297 @@
+                path = [path stringByAppendingPathComponent:BOXURLSessionTaskCacheResponseData];
+                break;
+            case BOXURLSessionTaskCacheFileTypeError:
+                path = [path stringByAppendingPathComponent:BOXURLSessionTaskCacheError];
+                break;
+        }
+
+        success = [self cacheAndEncryptData:data atPath:path error:&error];
+    }
+
+    if (outError != nil) {
+        (*outError) = error;
+    }
+    return success;
+}
+
+- (BOOL)cacheAndEncryptData:(NSData *)data atPath:(NSString *)path error:(NSError **)outError
+{
+    NSData *finalData = data;
+    if ([self.delegate respondsToSelector:@selector(encryptData:)]) {
+        finalData = [self.delegate encryptData:data];
+    }
+    return [finalData writeToFile:path options:NSDataWritingWithoutOverwriting error:outError];
+}
+
+// create if not exists, users/$userId/$associateId/$backgroundSessionId-$sessionTaskId
+- (BOOL)createFileForUserId:(NSString *)userId associateId:(NSString *)associateId backgroundSessionId:(NSString *)backgroundSessionId sessionTaskId:(NSUInteger)sessionTaskId error:(NSError **)error
+{
+    NSString *path = [self filePathOfUserSessionTaskGivenUserId:userId associateId:associateId backgroundSessionId:backgroundSessionId sessionTaskId:sessionTaskId];
+    return [self createFile:path error:error];
+}
+
+// create if not exists, sessions/$backgroundSessionId/$sessionTaskId
+- (BOOL)createDirForBackgroundSessionId:(NSString *)backgroundSessionId sessionTaskId:(NSUInteger)sessionTaskId error:(NSError **)error
+{
+    NSString *path = [self dirPathOfSessionTaskWithBackgroundSessionId:backgroundSessionId sessionTaskId:sessionTaskId];
+    return [self createDirectory:path error:error];
+}
+
+- (BOOL)createDirectory:(NSString *)path error:(NSError **)error
+{
+    BOOL isDir = NO;
+    BOOL success = YES;
+    if ([[NSFileManager defaultManager] fileExistsAtPath:path isDirectory:&isDir] == NO || isDir == NO) {
+        success = [[NSFileManager defaultManager] createDirectoryAtPath:path withIntermediateDirectories:YES attributes:nil error:error];
+    }
+    return success;
+}
+
+- (BOOL)deleteDirectory:(NSString *)directoryPath error:(NSError **)error
+{
+    BOOL isDir = NO;
+    BOOL success = YES;
+    if ([[NSFileManager defaultManager] fileExistsAtPath:directoryPath isDirectory:&isDir] == YES && isDir == YES) {
+        success = [[NSFileManager defaultManager] removeItemAtPath:directoryPath error:error];
+    }
+    return success;
+}
+
+- (BOOL)createFile:(NSString *)path error:(NSError **)error
+{
+    BOOL success = [[NSFileManager defaultManager] createFileAtPath:path contents:nil attributes:nil];
+    if (success == NO && error != nil) {
+        (*error) = [[NSError alloc] initWithDomain:BOXContentSDKErrorDomain code:BOXContentSDKSCacheErrorFileCreateFailed userInfo:nil];
+    }
+    return success;
+}
+
+// path sessions/$backgroundSessionId/$sessionTaskId
```

Comment:
MInor: can we properly caps/format these short method comments? e.g. `// Filepath Pattern: ...`